### PR TITLE
[ARM] az deployment sub/group what-if: Fix array alignment and error handling

### PR DESCRIPTION
**Pull request by shenglol**
**Description**  
We found a couple of small issues in the newly added `az deployment sub/group what-if` commands. The PR contains fixes for them:

- Fix a typo in a parameter help message
- Fix alignment for non-empty array in What-If results
- Added error handling for what-if error with 200 status code

**Testing Guide**  
To execute a resource group level what-if, run:
```bash
az deployment group what-if --template-file <path-to-template-file>
```

To execute a subscription level what-if, run:
```bash
az deployment sub what-if --template-file <path-to-template-file>
```

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S169

**Comment by shenglol** [comment]
Hi @zhoxing-ms Could you help take a look when you get a chance?

File: src/azure-cli/azure/cli/command_modules/resource/custom.py [hunk 1]
```diff
@@ -635,6 +635,13 @@ def what_if_deploy_arm_template_at_subscription_scope(cmd,
...
 
+    if what_if_result.error:
+        # The status code is 200 even when there's an error, because
+        # it is technically a successful What-If operation. The error
+        # is on the ARM template but not the operation.
+        err_message = _build_what_if_error_message(what_if_result.error)
+        raise CLIError(err_message)
+
     if no_pretty_print:
         return what_if_result
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/custom.py:646]: What about the error in the `what_if_poller` operation itself? How is this situation handled? Will exceptions be thrown?

Comment (shenglol) [reply src/azure-cli/azure/cli/command_modules/resource/custom.py:646]: Yes, exceptions will be thrown and handled by `handle_template_based_exception` specified in the command definitions.

**Comment by shenglol** [comment]
Hello @Juliehzl and @qianwens, could one of you take a look this PR?

**Pull request merged**